### PR TITLE
Miscellaneous fixes of ops | fix(torchlib)

### PR DESCRIPTION
**Pull request by justinchuby**
Fixes for comparison ops, mish, bitwise_shift from findings in https://github.com/justinchuby/torch-onnx-op-matrix/pull/20

### CODE REVIEW

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 1]
```diff
@@ -1086,10 +1089,47 @@ def aten_bitwise_and(self: TInt, other: TInt) -> TInt:
 
 
 @torch_op("aten::bitwise_left_shift")
-def aten_bitwise_left_shift(self: TInt, other: TInt) -> TInt:
+def aten_bitwise_left_shift_int16(self: INT16, other: INT16) -> INT16:
```
Comment (titaiwangms) [inline onnxscript/function_libs/torch_lib/ops/core.py:1092]: I forgot about this. When we call `op_schema.type_constraints`, do we get `tensor{int16}` from it?

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py [hunk 2]
```diff
@@ -732,9 +745,9 @@ def _where_input_wrangler(
     ),
     TorchLibOpInfo("gather", core_ops.aten_gather),
     TorchLibOpInfo("ge", core_ops.aten_ge),
-    # TorchLibOpInfo("greater_equal", core_ops.aten_greater_equal),  # no test case in OPS_DB
```
Comment (titaiwangms) [inline onnxscript/tests/function_libs/torch_lib/ops_test_data.py:735]: Do we need to add bool test case?

**Review by titaiwangms — APPROVED** [review]

File: onnxscript/function_libs/torch_lib/ops/core.py (hunk 1 above)
Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/core.py:1092]: I think so!

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py (hunk 2 above)
Comment (justinchuby) [reply onnxscript/tests/function_libs/torch_lib/ops_test_data.py:735]: They are added as ge_bool etc. Am I missing something?

**Comment by justinchuby** [comment]
Need to fix bitshift

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py (hunk 2 above)
Comment (titaiwangms) [reply onnxscript/tests/function_libs/torch_lib/ops_test_data.py:735]: I asked because I don't see extra_opinfo.py is modified.

Comment (justinchuby) [reply onnxscript/tests/function_libs/torch_lib/ops_test_data.py:735]: I see. I just duplicated the test. They include test cases with bool inputs.

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 3]
```diff
@@ -1116,10 +1156,98 @@ def aten_bitwise_or(self: TInt, other: TInt) -> TInt:
...
+
+@torch_op("aten::bitwise_right_shift")
+def aten_bitwise_right_shift_int32(self: INT32, other: INT32) -> INT32:
+    """bitwise_right_shift.Tensor(Tensor self, Tensor other) -> Tensor"""
+    negative = op.Less(self, 0)
+    self = op.Cast(self, to=UINT32.dtype)
+    other = op.Cast(other, to=UINT32.dtype)
+
+    # Simulate signed bitshift with unsigned bitshift
+    # Clear the lower bits for the mask by shifting right then left
```
Comment (gramalingam) [inline onnxscript/function_libs/torch_lib/ops/core.py:1189]: Can you clarify (preferably in comments in code) the desired behavior? I guess there are just two options, right, either preserving the sign bit or using zeros?

Comment (gramalingam) [reply onnxscript/function_libs/torch_lib/ops/core.py:1189]: Are you trying to simulate arithmetic shift using logical shift?

Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/core.py:1189]: > Are you trying to simulate arithmetic shift using logical shift?

Yes. What you said is more accurate

File: onnxscript/function_libs/torch_lib/ops/core.py (hunk 1 above)
Comment (gramalingam) [reply onnxscript/function_libs/torch_lib/ops/core.py:1092]: Is `other` expected to be a non-negative number (in spite of the typing)? Hope we don't need to handle negative shifts?

File: onnxscript/function_libs/torch_lib/ops/core.py (hunk 3 above)
Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/core.py]: Hopefully it's clearer now?

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 4]
```diff
@@ -1116,10 +1156,98 @@ def aten_bitwise_or(self: TInt, other: TInt) -> TInt:
...
+    mask = op.BitShift(
+        op.Cast(op.Constant(value_int=0xFFFFFFFF), to=UINT32.dtype), other, direction="LEFT"
+    )
+    mask = op.BitShift(mask, other, direction="RIGHT")
+    mask = op.BitwiseNot(mask)
+    # Do unsigned shift
+    shifted = op.BitShift(self, other, direction="RIGHT")
```
Comment (gramalingam) [inline onnxscript/function_libs/torch_lib/ops/core.py:1197]: Not sure why you have so many shifts? We need to shift 0xFFF... once to right by other once, and do a bitwise not, right?

File: onnxscript/function_libs/torch_lib/ops/core.py (hunk 1 above)
Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/core.py:1092]: That's a good point. Currently all tests pass so I assume so? I will add a comment

Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/core.py:1092]: Done

File: onnxscript/function_libs/torch_lib/ops/core.py (hunk 4 above)
Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/core.py:1199]: Thanks 😅 Done.

**Pull request merged**